dev/lint-python

```diff
@@ -122,6 +122,31 @@ function pycodestyle_test {
     fi
 }
 
+function mypy_test {
+    local MYPY_REPORT=
+    local MYPY_STATUS=
+
+    if ! hash "$MYPY_BUILD" 2> /dev/null; then
+        echo "The $MYPY_BUILD command was not found. Skipping MyPy test check build for now."
+        echo
+        return
+    fi
+
+    echo "starting $MYPY_BUILD test..."
+    MYPY_REPORT=$( ($MYPY_BUILD --config-file python/mypy.ini python/) 2>&1)
```

Comment:
I'd probably test `python/pyspark` directly and avoid adding `type: ignore` in `setup.py` and supporting fiiles.

Reply:
Much better indeed, thanks!